### PR TITLE
Provide a new ConvertV1Profiles class for converting V1 profiles

### PR DESCRIPTION
**Pull request by gejohnston**
**What It Does**

The new ConvertV1Profiles class enables apps to convert V1 profiles into a current Zowe client configuration file.
- Refactored logic from convert-profiles.handler and ConfigBuilder.convert into ConvertV1Profiles.convert.
- Removed ConfigBuilder.convert.
- Replaced IConfigConvertResult with IConvertV1Profiles (which contains IConvertV1ProfResult).
- Renamed class V1ProfileConversion (previously known as ProfileIO) to V1ProfileRead for clarity.
  - Marked class V1ProfileRead as @internal.

The primary function of the `convert profiles` command remains the same. The following changes were introduced:
- The command no longer prompts if you want to convert your profiles.
- The prompt asking if you want to delete your old profiles occurs before any detection of profiles and before any other information is displayed.
- Messages about all completed actions are displayed first, followed by messages for actions that failed.
- Some messages were stated more generically so that they are appropriate for both CLI and GUI apps.

The ConvertV1Profiles.convert API was written to work better for apps that are not a command line program.
- Instead of displaying messages to the console, the API places each line of message text into result object, along with formatting hints. Informational report messages are differentiated from error messages. By using the result object, all calling apps can display the same messages.
- The previous API threw exceptions in many error situations. It expected that the exception would just be displayed in a terminal. The new API catches exceptions and places error messages into the result object.

**How to Test**

The impact on end-users can be tested by running the `convert profiles` command in various scenarios. Because V1 profiles can no longer be created in V3, it can be difficult to create useful test data. The following screenshots demonstrate several known scenarios.

In a simple situation the output looks like the screenshot below. In this scenario, the V1 SCS plugin was already installed, and the `convert profiles` command uninstalled that plugin.

image

When you also ask to delete your old profiles, you are prompted to confirm the deletion. The output looks like this:

image

When an error occurs, the output looks like this:

image

When the old V1 SCS plugin is configured as the CredentialManager for Zowe, the .zowe\settings\imperative.json looks like this:

```
{
  "overrides": {
    "CredentialManager": "@zowe/secure-credential-store-for-zowe-cli"
  }
}
```

When configured like this, **ALL** zowe commands will display a warning that the Credential Manager is not in the list of known credential managers. The command then attempts its operation. An example of a `zowe-files list` command shows this behavior.

image

Also, when configured like this, the `plugins uninstall @zowe/secure-credential-store-for-zowe-cli` command (like all Zowe commands) displays the same CredentialManager warning. It also display a set of NPM warnings when uninstalling the V1 SCS plugin, even though it successfully uninstalls the SCS plugin.

image

The warnings above are not a result of the `convert profiles` command, but they do affect the `convert profiles` command because:
1. It is a zowe command and all zowe commands get the CredentialManager warnings in this scenario.
2. It automatically uninstalls the SCS plugin, so it gets the NPM warnings that occur from the `plugins uninstall` operation.

Despite the warnings in such a configuration, the `convert profiles` command converts the V1 profiles, uninstalls the SCS plugin, and restores the Zowe built-in CredentialManager.

image

As proof that the `convert profiles` command has restored the built-in CredentialManager, when you now run the `zos-files list` command, you no longer get the CredentialManager warning.

image

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Review by awharn — CHANGES_REQUESTED** [review]
Requesting changes due to several unused variable warnings and a DCO error on commit [8f24954](https://github.com/zowe/zowe-cli/pull/2113/commits/8f249541d6118438369db6130b91953e9a4f64e5) that requires attention.

**Review by anaxceron — APPROVED** [review]
👍

**Review by traeok — APPROVED** [review]
Implementation LGTM, thanks @gejohnston! I'll run some local tests to ensure the behavior is as expected and will update my review if something comes up.

File: packages/imperative/src/config/src/doc/IConvertV1Profiles.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+export interface IConvertV1ProfOpts {
+    // Should V1 profiles be deleted after conversion?
+    deleteV1Profs: boolean;
+}
+
+/**
+ * Message formatting constants. They can be ORed into IConvertMsg.msgFormat.
+ * Do not combine REPORT_LINE and ERROR_LINE.
+ */
+/* eslint-disable @typescript-eslint/no-magic-numbers */
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/imperative/src/config/src/doc/IConvertV1Profiles.ts:24]: The lint rule was left over from an earlier version of ConvertMsgFmt that was implemented as a class. The class needed the lint rule. After changing it to an enum no lint rule is needed, but I forgot to delete the lint rule. I will clean that up. Thanks for noticing.

File: packages/imperative/src/config/src/ConvertV1Profiles.ts [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+     */
+    private static async checkZoweKeyRingAvailable(): Promise<boolean> {
+        let success: boolean = false;
+        try {
+            const zoweSecretsPath = require.resolve("@zowe/secrets-for-zowe-sdk");
+            ConvertV1Profiles.zoweKeyRing = (await import(zoweSecretsPath)).keyring;
+            await ConvertV1Profiles.zoweKeyRing.findCredentials(CredentialManagerOverride.DEFAULT_CRED_MGR_NAME);
+            success = true;
+        } catch (err) {
+            success = false;
```
[the remark this answers is not in the document]
Comment (gejohnston) [reply packages/imperative/src/config/src/ConvertV1Profiles.ts:617]: This logic comes from the original function that I moved. I tend not change logic during a refactoring unless it is needed by the enhancement - just to reduce the liklihood of introducing a new error.

I did change a couple of unrelated items in this function. The code is a lot less volatile at this point. I plan to change the earlier lint rule anyway. So, I may as well clean this up. Although . . . my variation on your suggestion will be to return false in the catch block and return true at the end of the function - an identical result, but I just like the look of it better.

**Review by awharn — APPROVED** [review]

**Pull request merged**